extension/flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicOutputFormat.java

```diff
@@ -0,0 +1,120 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.doris.flink.table;
+
+import org.apache.doris.flink.cfg.DorisOptions;
+import org.apache.flink.api.common.io.RichOutputFormat;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+
+import java.io.IOException;
+import java.util.StringJoiner;
+
+/**
+ * DorisDynamicOutputFormat
+ **/
+public class DorisDynamicOutputFormat extends RichOutputFormat<RowData>  {
+
+
+    private  DorisOptions options ;
+    private DorisStreamLoad dorisStreamLoad;
+    private final String fieldDelimiter = "\t";
+
+    public DorisDynamicOutputFormat(DorisOptions options) {
+        this.options = options;
+    }
+
+    @Override
+    public void configure(Configuration configuration) {
+    }
+
+    @Override
+    public void open(int taskNumber, int numTasks) throws IOException {
+        dorisStreamLoad = new DorisStreamLoad(
+                options.getFenodes(),
+                options.getTableIdentifier().split("\\.")[0],
+                options.getTableIdentifier().split("\\.")[1],
+                options.getUsername(),
+                options.getPassword());
+        System.out.println(dorisStreamLoad);
+    }
+
+    @Override
+    public void writeRecord(RowData row) throws IOException {
```

Comment:
We need to buffer a batch of rows before triggering stream load

Reply:
yes，already submitted